spyderlib/utils/introspection/fallback_plugin.py

```diff
@@ -44,17 +46,34 @@ def get_completions(self, info):
             items += module_completion(info.line, [info.filename])
             return [(i, '') for i in sorted(items)]
         elif info.obj:
-            base = info.obj
-            tokens = set(info.split_words(-1))
-            items = [item for item in tokens if
-                     item.startswith(base) and len(item) > len(base)]
-            if '.' in base:
-                start = base.rfind('.') + 1
+            if info.context:
+                # get a list of token matches for the current object
+                tokens = info.lexer.get_tokens(info.source_code)
+                for (context, token) in tokens:
+                    token = token.strip()
+                    if (context in info.context and token.startswith(info.obj)
+                            and info.obj != token):
+                        items.append(token)
+                # add in keywords if not in a string
+                if context not in Token.Literal.String:
+                    try:
+                        keywords = list(info.lexer.tokens['keywords'][0][0].words)

```

Comment:
This is not working for some lexers I tried. For example, `BashLexer`, `JavaLexer` and `JavascriptLexer` don't have a `keywords` key as part of their tokens.

By the way, I'm using Pygments 2.0.2. Maybe the 1.x is different in that respect :-)